writeengine/splitter/we_cmdargs.cpp

```diff
@@ -217,10 +217,10 @@ std::string WECmdArgs::getCpImportCmdLine()
     if (fbTruncationAsError)
         aSS << " -S ";
 
-    if (!fS3Key.empty())
+    if (!fS3Key.empty() && !(fMode == 0 || fMode == 1))
     {
```

Comment:
these cmdline params shouldn't be propagated when this process is acting as a distributor.

Reply:
Also, since Mode 2 and S3 is not allowed, the only case these options are propagated to cpimport.bin is Mode 3?

Reply:
@roman yeah I saw references to mode 0 in the code, I'm not sure what that is or was.  I think it is an alias for mode 1 now, but might have originally had a different purpose.

@gagan Yes that's right.  Mode 1 means the splitter distributes data to a cpimport process on each PM.  Mode 2 means the splitter tells the cpimport processes to import local file F, which they all have by external means (ETL dropping a file with the same name on each node).  Obviously having them all fetch the same file from S3 doesn't make sense.  Mode 3 is a single-node import.  On a single-node cluster all modes are effectively mode 3, and so the args for importing from s3 need to be propagated,

Reply:
@pleblanc1976 This is very helpful, thanks for the clarification.